src/core_plugins/timelion/public/panels/timechart/schema.js

```diff
@@ -274,6 +281,8 @@ export default function timechartFn(Private, config, $rootScope, timefilter, $co
           _.each(canvasElem.find('.ngLegendValue'), function (elem) {
             $compile(elem)(legendScope);
           });
+
+          legendTable = canvasElem.find('div.legend table');
```

Comment:
Could we instead of destroying and creating the `caption` element on each mouse move event (which I would consider lots of unnecessary DOM operations), create it once here, store the reference, and just update the content of it when hovering the graph?